contracts/NumeraireDelegateV4.sol

```diff
@@ -0,0 +1,186 @@
+pragma solidity ^0.4.23;
+
+import "contracts/StoppableShareable.sol";
+import "contracts/NumeraireShared.sol";
+
+// Whoever creates the contract has the power to stop it, this person can be changed via transferOwnership(_new_address)
+contract NumeraireDelegate is StoppableShareable, NumeraireShared {
+
+    function NumeraireDelegate(address[] _owners, uint256 _num_required) StoppableShareable(_owners, _num_required) {
+    }
+
+    // All minted NMR are initially sent to Numerai, obeying both weekly and total supply caps
+    function mint(uint256 _value) onlyOwner returns (bool ok) {
+        // Prevent minting more than the supply cap.
+        require(safeAdd(total_minted, _value) <= supply_cap);
+
+        // Prevent minting more than the disbursement.
+        require(_value <= getMintable());
+
+        balanceOf[numerai] = safeAdd(balanceOf[numerai], _value);
+        totalSupply = safeAdd(totalSupply, _value);
+        total_minted = safeAdd(total_minted, _value);
+
+        // Notify anyone listening.
+        Mint(_value);
+
+        return true;
+    }
+
+    /* NOTE: this violates the invariant that releaseStake or destroyStake are only called once.
+        - StakeReleased event will emit _releaseAmount instead of etherReward
+        - StakeReleased evetn can be called multiple times per round
+    */
+
+    // Numerai calls this function to release staked tokens when the staked predictions were successful
+    function releaseStake(address _staker, bytes32 _tag, uint256 _releaseAmount, uint256 _tournamentID, uint256 _roundID, bool _successful) onlyOwner stopInEmergency returns (bool ok) {
+        var round = tournaments[_tournamentID].rounds[_roundID];
+        var stake = round.stakes[_staker][_tag];
+        var originalStakeAmount = stake.amount;
+
+        require(stake.amount > 0);
+        require(!stake.resolved);
+        require(round.resolutionTime <= block.timestamp);
+
```

Comment:
can we add something like
```
if (_releaseAmount == 0) {
            _releaseAmount = stake.amount
        }
```

We want control for releasing a partial amount, but I also don't want to need to know the exact amount to 0 out the stake and risk leaving ETH unreleased and it needs to be 0 to set resolved to true.